File: src/main/java/net/liyze/basin/core/commands/RemoteCommand.java
package net.liyze.basin.core.commands;

import net.liyze.basin.context.annotation.Component;
import net.liyze.basin.remote.Client;
import net.liyze.basin.core.Command;
import org.jetbrains.annotations.NotNull;

import java.util.List;

import static net.liyze.basin.core.Main.LOGGER;
import static net.liyze.basin.core.Main.envMap;

@Component
public class RemoteCommand implements Command {
    @Override
    public void run(@NotNull List<String> args) {
        String host = args.remove(0);
        try {
            Client.send(String.join(" ", args), host, envMap.get("\"" + host + "_token\"").toString());
        } catch (Exception e) {
            LOGGER.error(e.toString());
        }
    }

    @Override
    public @NotNull String Name() {
        return "remote";
    }
}

File: src/main/kotlin/net/liyze/basin/core/scan/RemoteCommand.kt
/*
 * Copyright (c) 2023 Liyze09
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.liyze.basin.core.scan

import net.liyze.basin.common.printException
import net.liyze.basin.core.Command
import net.liyze.basin.core.envMap
import net.liyze.basin.core.remote.send
import org.slf4j.Logger


class RemoteCommand : Command {
    override fun run(args: List<String?>, logger: Logger) {
        val host: String = args[0]!!
        try {
            envMap["\"" + host + "_token\""]?.let {
                send(
                    java.lang.String.join(" ", args.subList(1, args.size)),
                    host,
                    it
                )
            } ?: throw RuntimeException("Remote token not found: $host")
        } catch (e: Exception) {
            e.printException()
        }
    }

    override fun getName(): String {
        return "remote"
    }
}
